ui/js/dfv/src/fields/number-field/index.js

```diff
@@ -1,33 +1,131 @@
-/* eslint-disable react/prop-types */
 import React from 'react';
-import BaseInput from 'dfv/src/fields/base-input';
-// import * as validationRules from 'dfv/src/validation/validation-rules';
-
-// @todo this is an incomplete field component
-// @todo add tests?
-const NumberField = ( props ) => {
-	// noinspection JSUnresolvedVariable
-	/* Enable this only for slider
-	props.validation.addRules( [
-		{
-			rule: validationRules.max( props.value, props.fieldConfig.number_max ),
-			condition: true,
-		},
-		{
-			rule: validationRules.min( props.value, props.fieldConfig.number_min ),
-			condition: true,
-		},
-	] );*/
-
-	// noinspection JSUnresolvedVariable
+import { debounce } from 'lodash';
+import PropTypes from 'prop-types';
+
+import numberFormat from 'dfv/src/helpers/numberFormat';
+import { toBool } from 'dfv/src/helpers/booleans';
+import { FIELD_PROP_TYPE_SHAPE } from 'dfv/src/config/prop-types';
+
+import './number-field.scss';
+
+const formatValue = ( newValue, decimals, format ) => {
```

Comment:
@zrothauser we should move this into a helper function that will be shared between both currency and number fields

Reply:
Fixed on my side